### ISSUE TITLE
Support Go proxied struct methods returning non-maps

### ISSUE THREAD

**Issue by jakegut** [risor-io/risor#79]
Using embedded risor, I'd like the method of a proxied Go struct to return either:

a. A struct with additional methods to call in risor:
<details>

<summary>Example A</summary>

```go
package main

import (
	"context"
	"flag"
	"fmt"
	"os"

	"github.com/fatih/color"
	"github.com/risor-io/risor"
	"github.com/risor-io/risor/object"
)

type State struct {
	Name    string
	Running bool
}

type Service struct {
	name       string
	running    bool
	startCount int
	stopCount  int
}

func (s *State) IsRunning() bool {
	return s.Running
}

func (s *Service) Start() error {
	if s.running {
		return fmt.Errorf("service %s already running", s.name)
	}
	s.running = true
	s.startCount++
	return nil
}

func (s *Service) Stop() error {
	if !s.running {
		return fmt.Errorf("service %s not running", s.name)
	}
	s.running = false
	s.stopCount++
	return nil
}

func (s *Service) SetName(name string) {
	s.name = name
}

func (s *Service) GetName() string {
	return s.name
}

func (s *Service) PrintState() {
	fmt.Printf("printing state... name: %s running %t\n", s.name, s.running)
}

func (s *Service) GetState() State {
	return State{
		Name:    s.name,
		Running: s.running,
	}
}

const defaultExample = `
svc.SetName("My Service")
svc.Start()
state := svc.GetState()
state.IsRunning()
`

var red = color.New(color.FgRed).SprintfFunc()

func main() {
	var code string
	flag.StringVar(&code, "code", defaultExample, "Code to evaluate")
	flag.Parse()

	ctx := context.Background()

	// Initialize the service
	svc := &Service{}

	// Create a Risor proxy for the service
	proxy, err := object.NewProxy(svc)
	if err != nil {
		fmt.Println(red(err.Error()))
		os.Exit(1)
	}

	// Build up options for Risor, including the proxy as a variable named "svc"
	opts := []risor.Option{
		risor.WithDefaultBuiltins(),
		risor.WithBuiltins(map[string]object.Object{"svc": proxy}),
	}

	// Run the Risor code which can access the service as `svc`
	if _, err = risor.Eval(ctx, code, opts...); err != nil {
		fmt.Println(red(err.Error()))
		os.Exit(1)
	}
}
```

</details>

b. An `object.Object` (that would really be an `*object.Proxy`) so that I can use the methods of the returned struct

<details>

<summary>Example B (this one stack overflows)</summary>

```go
package main

import (
	"context"
	"flag"
	"fmt"
	"os"

	"github.com/fatih/color"
	"github.com/risor-io/risor"
	"github.com/risor-io/risor/object"
)

type State struct {
	Name    string
	Running bool
}

type Service struct {
	name       string
	running    bool
	startCount int
	stopCount  int
}

func (s *State) IsRunning() bool {
	return s.Running
}

func (s *Service) Start() error {
	if s.running {
		return fmt.Errorf("service %s already running", s.name)
	}
	s.running = true
	s.startCount++
	return nil
}

func (s *Service) Stop() error {
	if !s.running {
		return fmt.Errorf("service %s not running", s.name)
	}
	s.running = false
	s.stopCount++
	return nil
}

func (s *Service) SetName(name string) {
	s.name = name
}

func (s *Service) GetName() string {
	return s.name
}

func (s *Service) PrintState() {
	fmt.Printf("printing state... name: %s running %t\n", s.name, s.running)
}

func (s *Service) GetState() object.Object {
	state := &State{
		Name:    s.name,
		Running: s.running,
	}
	stateProxy, _ := object.NewProxy(state)
	return stateProxy
}

const defaultExample = `
svc.SetName("My Service")
svc.Start()
state := svc.GetState()
state.IsRunning()
`

var red = color.New(color.FgRed).SprintfFunc()

func main() {
	var code string
	flag.StringVar(&code, "code", defaultExample, "Code to evaluate")
	flag.Parse()

	ctx := context.Background()

	// Initialize the service
	svc := &Service{}

	// Create a Risor proxy for the service
	proxy, err := object.NewProxy(svc)
	if err != nil {
		fmt.Println(red(err.Error()))
		os.Exit(1)
	}

	// Build up options for Risor, including the proxy as a variable named "svc"
	opts := []risor.Option{
		risor.WithDefaultBuiltins(),
		risor.WithBuiltins(map[string]object.Object{"svc": proxy}),
	}

	// Run the Risor code which can access the service as `svc`
	if _, err = risor.Eval(ctx, code, opts...); err != nil {
		fmt.Println(red(err.Error()))
		os.Exit(1)
	}
}
```

</details>

Of course, I could alter the underlying Go structs to get around this. However, I was wondering if it would be possible to do either A or B (or another option that's already supported) to get around this?

**Comment by myzie** [comment]
Good catch @jakegut, agreed. I started taking a look at this and it should be possible. I'm aiming to get something like your "example A", where nested structs are also proxied as appropriate. 

I'll keep you updated.

**Comment by myzie** [comment]
I think I have this working on the branch `proxy-improvements`:
https://github.com/risor-io/risor/pull/81

If you want to try it out, check out that branch and run `go run ./examples/go/struct`.

**Comment by jakegut** [comment]
Looks great!

**Comment by myzie** [comment]
This is now available in the v0.13.0 release, and on the editor on https://risor.io.
